eventdemo/source/objc/EventDemo/ViewControllers/RoomViewController.m

```diff
@@ -0,0 +1,89 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#import "RoomViewController.h"
+#import "ChatClientManager.h"
+#import "MessageCell.h"
+
+
+@implementation RoomViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    self.title = _roomName;
+    _tableView.rowHeight = UITableViewAutomaticDimension;
+    _tableView.estimatedRowHeight = 44;
+
+    [self setupNotifications];
+    [self updateMessages];
+}
+
+- (void)updateMessages
+{
+    self.messages = [[ChatClientManager sharedManager] messagesForRoom:_roomName];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.tableView reloadData];
+    });
+}
+
+- (IBAction)sendPressed:(id)sender
+{
+    [[ChatClientManager sharedManager] sendMessage:_messageTextField.text
+                                              room:_roomName];
+    _messageTextField.text = @"";
+}
+
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
```

Comment:
Please separate UITableViewDelegate and UITableViewDataSource with corresponding pragma marks.